hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/WritableFileStoreImpl.java

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.file.impl;
+
+import static com.hedera.node.app.service.file.impl.FileServiceImpl.BLOBS_KEY;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.state.file.File;
+import com.hedera.node.app.service.mono.state.merkle.MerkleTopic;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.node.app.spi.state.WritableKVState;
+import com.hedera.node.app.spi.state.WritableKVStateBase;
+import com.hedera.node.app.spi.state.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Provides write methods for modifying underlying data storage mechanisms for
+ * working with Files.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ * This class is not complete, it will be extended with other methods like remove, update etc.,
+ */
+public class WritableFileStoreImpl extends ReadableFileStoreImpl {
+    /** The underlying data storage class that holds the file data. */
+    private final WritableKVState<EntityNum, File> filesState;
+
+    /**
+     * Create a new {@link WritableFileStoreImpl} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableFileStoreImpl(@NonNull final WritableStates states) {
+        super(states);
+        this.filesState = Objects.requireNonNull(states.get(BLOBS_KEY));
+    }
+
+    /**
+     * Persists a new {@link File} into the state, as well as exporting its ID to the transaction
+     * receipt.
+     *
+     * @param file - the file to be mapped onto a new {@link MerkleTopic} and persisted.
+     */
+    public void put(@NonNull final File file) {
+        filesState.put(EntityNum.fromLong(Objects.requireNonNull(file).fileNumber()), file);
+    }
+
+    /**
+     * Commits the changes to the underlying data storage.
+     * TODO: Not sure if the stores have responsibility of committing the changes. This might change in the future.
+     */
+    public void commit() {
+        if (filesState instanceof WritableKVStateBase) ((WritableKVStateBase) filesState).commit();
+    }
+
+    /**
+     * Returns the {@link File} with the given number. If no such file exists, returns {@code Optional.empty()}
+     * @param fileNum - the number of the file to be retrieved.
+     */
+    public @Nullable Optional<File> get(final long fileNum) {
+        requireNonNull(fileNum);
+        final var file = filesState.get(EntityNum.fromLong(fileNum));
+        return Optional.ofNullable(file);
+    }
+
+    /**
+     * Returns the {@link File} with the given number using {@link WritableKVState#getForModify(Comparable K)}.
```

Comment:
There's a javadoc error here - 

/runner/_work/hedera-services/hedera-services/hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/WritableFileStoreImpl.java:84: error: reference not found
     * Returns the {@link File} with the given number using {@link WritableKVState#getForModify(Comparable K)}.

Reply:
I fixed, I didn't commit it yet thanks